libbeat/processors/decode_xml/decode_xml_test.go

```diff
@@ -0,0 +1,482 @@
+						"book": []interface{}{
+							map[string]interface{}{
+								"author": "William H. Gaddis",
+								"review": "One of the great seminal American novels of the 20th century.",
+								"title":  "The Recognitions",
+							},
+							map[string]interface{}{
+								"author": "Ralls, Kim",
+								"review": "Some review.",
+								"title":  "Midnight Rain",
+							},
+						},
+						"secondcategory": map[string]interface{}{
+							"paper": map[string]interface{}{
+								"description": "A former architect battles corporate zombies, \n\t\t\t\t\t\tan evil sorceress, and her own childhood to become queen \n\t\t\t\t\t\tof the world.",
+								"id":          "bk102",
+								"test2":       "Ralls, Kim",
+							},
+						},
+					},
+				},
+			},
+			error:        false,
+			errorMessage: "",
+		},
+		{
+			description: "Decoding with broken XML format, with IgnoreFailure false",
+			config: decodeXMLConfig{
+				Field:         "message",
+				IgnoreFailure: false,
+			},
+			Input: common.MapStr{
+				"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+						<author>William H. Gaddis</author>
+						<title>The Recognitions</title>
+						<review>One of the great seminal American novels of the 20th century.</review>
+				</ook>
+				catalog>`,
+			},
+			Output: common.MapStr{
+				"message": (map[string]interface{})(nil),
+				"error":   "failed to decode fields in decode_xml processor: error decoding XML field: xml.Decoder.Token() - XML syntax error on line 7: element <book> closed by </ook>",
+			},
+			error:        true,
+			errorMessage: "failed to decode fields in decode_xml processor: error decoding XML field: xml.Decoder.Token() - XML syntax error on line 7: element <book> closed by </ook>",
+		},
+		{
+			description: "Decoding with broken XML format, with IgnoreFailure true",
+			config: decodeXMLConfig{
+				Field:         "message",
+				IgnoreFailure: true,
+			},
+			Input: common.MapStr{
+				"message": `<?xml version="1.0"?>
+				<catalog>
+					<book>
+						<author>William H. Gaddis</author>
+						<title>The Recognitions</title>
+						<review>One of the great seminal American novels of the 20th century.</review>
+				</ook>
+				catalog>`,
+			},
+			Output: common.MapStr{
+				"message": (map[string]interface{})(nil),
+			},
+			error:        true,
+			errorMessage: "failed to decode fields in decode_xml processor: error decoding XML field: xml.Decoder.Token() - XML syntax error on line 7: element <book> closed by </ook>",
+		},
+		{
+			description: "Test when the XML field is empty, IgnoreMissing false",
+			config: decodeXMLConfig{
+				Field:         "message2",
+				IgnoreMissing: false,
+			},
+			Input: common.MapStr{
+				"message": "testing message",
+			},
+			Output: common.MapStr{
+				"message": "testing message",
+				"error":   "key not found; field value is not a string",
+			},
+			error:        true,
+			errorMessage: "key not found; field value is not a string",
```

Comment:
There are two errors here. I would expect only the `key not found` error. After that it should stop processing. Additionally it would nice to include the name of the missing key for context.